### ISSUE TITLE
Make survival AI wizards start with ranged attacks?

### ISSUE THREAD

**Issue by elidupree** [elidupree/EoHS#84]
The reason I don't give AI wizards full skills is that I want to leave the option of choosing skills open for a player who takes over the side. In survivals, there's no worry of that happening. The only question is how to detect that situation.

Another possibility is to allow you to retrain skills once if you  take over from an AI.

**Comment by momothefiddler** [comment]
I definitely see a benefit to allowing retraining after a control swap, but it also seems easy to trade a side and trade it back just to respec.

**Comment by elidupree** [comment]
Yeah, that's why I suggest that a wizard can only retrain once, and only if it was never controlled by a human before that.

Specifically, we wouldn't want to let the human and retrain after casting a spell, recruiting, attacking, or moving, which can all benefit from the previous skills. Maybe the skills could be automatically wiped as soon as the human takes over, although there are some difficulties if the control swap happens during the AI's turn.

A few more things to think about:

* The AI might eventually be able to actually CAST spells. However, this probably still wouldn't make retraining exploits be much use.
* It would also be nice to retrain if a newbie (or an asshole) picks bad skills and then leaves in the first couple of turns. Obviously, the game can't really detect this situation. There are some theoretical ways to require the other players' permission to retrain, but they have some UI inconveniences.
* What if you could, in general, retrain any skills you hadn't actually used yet? This is equivalent to just saving up your points, which you can already do, just with more UI inconvenience. However, you'd basically never be able to retrain alliances or rank 1-2 spells, because you theoretically are using them to boost your stats as soon as you end your turn.

**Comment by momothefiddler** [comment]
Oh! Well if you can track whether a side has _ever_ been controlled by a human, that seems like a reasonable way to handle it.

Retraining unused skills seems... nice in theory, but not nice enough to justify the inevitable confusion about what's been "used" - alliances and rank 1-2 spells, as mentioned, probably prerequisites even if you want to swap them for other prereqs, Enchanter even if you've only cast enchantments on turns that you've ended with unused AP >= 2*number of enchantments... I can imagine a lot of arguments over what qualifies as using a thing.
